src/util/TypeTraits.h

```diff
@@ -317,4 +317,8 @@ concept RegularInvocableWithExactReturnType =
 // auto&& x) // guaranteed rvalue reference, can safely be moved.
 template <typename T>
 concept Rvalue = std::is_rvalue_reference_v<T&&>;
+
+// Ensures that T is a floating-point type.
+template <typename T>
+CPP_concept FloatingPoint = std::is_floating_point_v<T>;
```

Comment:
I still have to figure out where to put them and what to name them.

I think I already have a file `src/backports/concepts.h` where I want a common namespace for the concepts that are in C++20.